include/zephyr/drivers/auxdisplay.h

```diff
@@ -0,0 +1,824 @@
+/*
+ * Copyright (c) 2022-2023 Jamie McCrae
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for auxiliary (textual/non-graphical) display drivers
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_AUXDISPLAY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_AUXDISPLAY_H_
+
+/**
+ * @brief Auxiliary (Text) Display Interface
+ * @defgroup auxdisplay_interface Text Display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/types.h>
+#include <zephyr/pm/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AUXDISPLAY_LIGHT_NOT_SUPPORTED 0
+
+/** @brief Used to describe the mode of an auxiliary (text) display */
+typedef uint32_t auxdisplay_mode_t;
+
+/** @brief Used for moving the cursor or display position */
+enum auxdisplay_position {
+	/** Moves to specified X,Y position */
+	AUXDISPLAY_POSITION_ABSOLUTE = 0,
+
+	/** Shifts current position by +/- X,Y position, does not take display direction into
+	 *  consideration
+	 */
+	AUXDISPLAY_POSITION_RELATIVE,
+
+	/** Shifts current position by +/- X,Y position, takes display direction into
+	 *  consideration
+	 */
+	AUXDISPLAY_POSITION_RELATIVE_DIRECTION,
+
+	AUXDISPLAY_POSITION_COUNT,
+};
+
+/** @brief Used for setting character append position */
+enum auxdisplay_direction {
+	/** Each character will be placed to the right of existing characters */
+	AUXDISPLAY_DIRECTION_RIGHT = 0,
+
+	/** Each character will be placed to the left of existing characters */
+	AUXDISPLAY_DIRECTION_LEFT,
+
+	AUXDISPLAY_DIRECTION_COUNT,
+};
+
+struct auxdisplay_light {
+	/** Minimum and maximum light levels, will both be 0 if unsupported */
+	uint8_t minimum;
+	uint8_t maximum;
+};
```

Comment:
this is not documented correctly (struct, each field)

Reply:
Fixed